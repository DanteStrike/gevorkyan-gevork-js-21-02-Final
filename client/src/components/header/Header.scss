@import "src/assets/mixins";
@import "src/assets/variables";

.header {
  background-color: var(--primary-background);
  border-bottom: var(--primary-border);

  &__container {
    @include container;

    display: flex;
    justify-content: space-between;
    min-height: 50px;
  }

  &__item {
    display: flex;
  }
}

@media (max-width: $tabletWidth) {
  .header {
    &__container {
      flex-direction: column;
      justify-content: flex-start;
    }

    &__item {
      justify-content: center;
      border-bottom: var(--primary-border);
    }

    &__item--logo {
      order: 1;
    }

    &__item--menu {
      order: 3;
    }

    &__item--auth {
      order: 2;
    }
  }
}
