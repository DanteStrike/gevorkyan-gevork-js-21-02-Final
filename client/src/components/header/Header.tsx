import React from 'react';
import './Header.scss';
import Logo from '../logo/Logo';
import Menu from '../menu/Menu';
import ConnectedAuth from '../connected-auth/ConnectedAuth';

function Header() {
  return (
    <header className="header">
      <div className="header__container">
        <div className="header__item header__item--logo">
          <Logo />
        </div>
        <div className="header__item header__item--menu">
          <Menu />
        </div>
        <div className="header__item header__item--auth">
          <ConnectedAuth />
        </div>
      </div>
    </header>
  );
}

export default React.memo(Header);
