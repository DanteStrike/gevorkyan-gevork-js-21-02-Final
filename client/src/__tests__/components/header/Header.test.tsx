import {shallow} from 'enzyme';
import React from 'react';
import Header from '../../../components/header/Header';

describe(`Component Header should render correctly`, () => {
  it(`snapshot default test`, () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });
});
