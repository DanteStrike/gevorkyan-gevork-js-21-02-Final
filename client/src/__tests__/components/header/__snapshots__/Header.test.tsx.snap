// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Header should render correctly snapshot default test 1`] = `
<header
  className="header"
>
  <div
    className="header__container"
  >
    <div
      className="header__item header__item--logo"
    >
      <Memo(Logo) />
    </div>
    <div
      className="header__item header__item--menu"
    >
      <Memo(Menu) />
    </div>
    <div
      className="header__item header__item--auth"
    >
      <Memo(ConnectedAuth) />
    </div>
  </div>
</header>
`;
